hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -179,7 +171,8 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
-        out.writeUTF(groupPassword);
+        //TODO @tkountis remove in 4.0
+        out.writeUTF(EMPTY_PWD);
```

Comment:
I think this could be worth a try. It shouldn't be more than `out.writeUtf(clusterService.getClusterVersion().isLessThan(Versions.V3_9) ? groupPassword : EMPTY_PWD);`, shouldn't it?